node/src/components/block_executor.rs

```diff
@@ -245,25 +286,38 @@ impl BlockExecutor {
     ) -> Effects<Event> {
         let next_deploy = match state.remaining_deploys.pop_front() {
             Some(deploy) => deploy,
-            None => {
-                // The state hash of the last execute-commit cycle is used as the block's post state
-                // hash.
-                let next_height = state.finalized_block.height() + 1;
-                let block = self.create_block(state.finalized_block, state.pre_state_hash);
-
-                let mut effects = effect_builder
-                    .announce_linear_chain_block(block, state.execution_results)
-                    .ignore();
-                // If the child is already finalized, start execution.
-                if let Some((finalized_block, deploys)) = self.exec_queue.remove(&next_height) {
-                    effects.extend(self.handle_get_deploys_result(
-                        effect_builder,
-                        finalized_block,
-                        deploys,
-                    ));
+            None => match state.finalized_block.era_end().as_ref() {
+                Some(era_end) => {
+                    let reward_items = era_end
+                        .rewards
+                        .iter()
+                        .map(|(&vid, &value)| RewardItem {
+                            validator_id: vid.into(),
+                            value,
+                        })
+                        .collect();
+                    let slash_items = era_end
+                        .equivocators
+                        .iter()
+                        .map(|&vid| SlashItem {
+                            validator_id: vid.into(),
+                        })
+                        .collect();
+                    let request = StepRequest {
+                        pre_state_hash: state.pre_state_hash.into(),
+                        protocol_version: ProtocolVersion::V1_0_0,
```

Comment:
Beyond the scope of this PR, but all the various places where ProtocolVersion is being hardcoded, it should be getting passed in.